### PR TITLE
Add MAX78002 SoC and MAX78002EVKIT board

### PR DESCRIPTION
**Pull request by hfakkiz**
The [MAX78002](https://www.analog.com/en/products/max78002.html) microcontroller (MCU) is an advanced system-on-chip (SoC) featuring an Arm® Cortex®-M4F and RISC-V CPU, large Flash and SRAM memories, and ultra-low-power deep neural network accelerator.

This PR adds MAX78002 and its EvKit board.

- Add [MAX78002](https://www.analog.com/en/products/max78002.html) SoC support
- Add [MAX78002EVKIT](https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/max78002evkit.html) support
- Enable tests for peripherals drivers

doc page in CI: https://builds.zephyrproject.io/zephyr/pr/80247/docs/boards/adi/max78002evkit/doc/index.html

**Tested with:**
_tests/drivers/uart/uart_basic_api_
uart_basic_api_test_ss

_tests/drivers/gpio/gpio_basic_api_
gpio_basic_api_test_ss

_tests/drivers/entropy/api_
trng_entropy_test_ss

_tests/drivers/watchdog/wdt_basic_api_
wdt_basic_api_test_ss

_tests/drivers/dma/chan_blen_transfer_
dma_chan_blen_transfer_test_ss

_tests/drivers/dma/loop_transfer_
dma_loop_transfer_test_ss

_tests/drivers/flash/common_
flash_common_test_ss

_tests/drivers/pwm/pwm_api_
pwm_api_test_ss

_tests/drivers/adc/adc_api_
adc_api_test_ss

_tests/drivers/spi/spi_loopback_
spi_loopback_test_ss

_tests/drivers/i2c/i2c_target_api_
i2c_target_api_test_ss

_tests/drivers/w1/w1_api_
w1_api_test_ss

_tests/drivers/counter/counter_basic_api_
counter_basic_api_test_sss

_samples/drivers/counter/alarm_
counter_alarm_sample_ss

### CODE REVIEW

File: boards/adi/max78002evkit/doc/img/max78002evkit_img1.jpg
Comment (nordicjm) [inline boards/adi/max78002evkit/doc/img/max78002evkit_img1.jpg:1]: Convert images to webp, then put through https://tinypng.com/

File: boards/adi/max78002evkit/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+  - Color TFT Display
+  - Audio Stereo Codec Interface
+  - Digital Microphone
+  - A 8Mb QSPI ram
+
+Supported Features
+==================
+
+Below are the interfaces supported by Zephyr on MAX78002EVKIT.
```
Comment (nordicjm) [inline boards/adi/max78002evkit/doc/index.rst:131]: needs board name as used with west build, surrounding with \`\` and called board target

File: boards/adi/max78002evkit/max78002evkit_max78002_m4.dts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			 zephyr,code = <INPUT_KEY_0>;
+		 };
+		 pb2: pb2 {
+			 gpios = <&gpio2 7 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			 label = "SW3";
+			 zephyr,code = <INPUT_KEY_1>;
+		 };
+		 pb_wakeup: pb_wakeup {
+			gpios = <&gpio3 1 (GPIO_PULL_UP | GPIO_ACTIVE_LOW
+								| MAX32_GPIO_VSEL_VDDIOH)>;
```
Comment (nordicjm) [inline boards/adi/max78002evkit/max78002evkit_max78002_m4.dts:51]: indent looks suspicious

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/adi/max78002evkit/doc/img/max78002evkit_img1.jpg
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/img/max78002evkit_img1.jpg:1]: Updated images, thanks.

File: boards/adi/max78002evkit/doc/index.rst (hunk 1 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/index.rst]: Updated file, thanks.

File: boards/adi/max78002evkit/max78002evkit_max78002_m4.dts (hunk 2 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/max78002evkit_max78002_m4.dts:51]: It's normal in code editor. I think it's about editors' tab view differences.

Comment (nordicjm) [reply boards/adi/max78002evkit/max78002evkit_max78002_m4.dts:51]: Zephyr tab is 8 spaces, the the spacing shown on github is correct, replaced with spaces would be:
```
                 pb_wakeup: pb_wakeup {
                        gpios = <&gpio3 1 (GPIO_PULL_UP | GPIO_ACTIVE_LOW
                                                                | MAX32_GPIO_VSEL_VDDIOH)>;
```

File: boards/adi/max78002evkit/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,297 @@
+.. _max78002_evkit:
+
+MAX78002EVKIT
+#############
```
Comment (kartben) [inline boards/adi/max78002evkit/doc/index.rst:4]: .. zephyr:board:: max78002_evkit

File: boards/adi/max78002evkit/board.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+board:
+  name: max78002evkit
```
Comment (kartben) [inline boards/adi/max78002evkit/board.yml:5]:
```suggestion
  name: max78002evkit
  full_name: MAX78002EVKIT
```

**Review by ttmut — CHANGES_REQUESTED** [review]

File: boards/adi/max78002evkit/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+generations of artificial intelligence (AI) products. The kit provides optimal versatility with a modular peripheral
+architecture, allowing a variety of input and output devices to be remotely located. DVP and CSI cameras,
+I2S audio peripherals, digital microphones, and analog sensors are supported, while a pair of industry-standard
+QWIIC connectors supports a large and growing array of aftermarket development boards. An onboard stereo
+audio codec offers line-level audio input and output, and tactile input is provided by a touch-enabled 2.4in TFT
+display. The MAX78002 energy consumption is tracked by a power accumulator, with four channels of formatted
+results presented on a secondary TFT display. All device GPIOs are accessible on 0.1in pin headers. A standard
+coaxial power jack serves as power input, using the included 5V, 3A wall-mount adapter. Two USB connectors
+provide serial access to the MAX78002, one directly and the other through a USB to UART bridge. A third
+USB connector allows access to the MAX78002 energy consumption data. Rounding out the features, a microSD
```
Comment (kartben) [inline boards/adi/max78002evkit/doc/index.rst:18]: this is a mouthful (but not your fault as it looks copy-pasted :))
please try to add a bit of breathing room by at least adding a blank line after first sentence (just as in https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/max78002evkit.html#eb-overview, basically)

File: boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp
Comment (kartben) [inline boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp:1]: should be named after the board name -- `max78002evkit.webp`

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/adi/max78002evkit/max78002evkit_max78002_m4.dts (hunk 2 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/max78002evkit_max78002_m4.dts]: Updated, thanks.

File: boards/adi/max78002evkit/Kconfig.max78002evkit [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# MAX78002EVKIT boards configuration
```
[the remark this answers is not in the document]
Comment (hfakkiz) [reply boards/adi/max78002evkit/Kconfig.max78002evkit]: Updated, thanks.

File: boards/adi/max78002evkit/max78002evkit_max78002_m4.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  - dma
+  - watchdog
+  - spi
+  - adc
+  - counter
+  - pwm
+  - w1
+  - flash
+ram: 384
+flash: 2500
```
[the remark this answers is not in the document]
Comment (hfakkiz) [reply boards/adi/max78002evkit/max78002evkit_max78002_m4.yaml]: Updated, thanks.

File: dts/arm/adi/max32/max78002.dtsi [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <DT_FREQ_M(25)>;
+			status = "disabled";
+		};
+	};
+};
+
+&flash0 {
+	reg = <0x10000000 DT_SIZE_K(2500)>;
```
[the remark this answers is not in the document]
Comment (hfakkiz) [reply dts/arm/adi/max32/max78002.dtsi]: Updated, thanks.

File: boards/adi/max78002evkit/doc/index.rst (hunk 3 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/index.rst]: Updated, thanks.

File: boards/adi/max78002evkit/board.yml (hunk 4 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/board.yml:5]: Updated, thanks.

File: boards/adi/max78002evkit/doc/index.rst (hunk 5 above)
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/index.rst]: Updated according to your suggestion, thanks.

File: boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp:1]: There are 2 board images, front and back. I updated image names as "max78002evkit_front.webp" and "max78002evkit.back". Is it okay for you?

File: boards/adi/max78002evkit/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,298 @@
+.. zephyr:board:: max78002_evkit
```
Comment (kartben) [inline boards/adi/max78002evkit/doc/index.rst:1]:
```suggestion
.. zephyr:board:: max78002evkit
```

apologies, I should have double checked in the board.yaml file

File: boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp
Comment (kartben) [reply boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp:1]: for the sake of indicating clearly which one is the "default", i would just name "max78002evkit_front.webp" as "max78002evkit.webp". Thanks!

Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/img/max78002evkit_img1.webp:1]: Updated as your suggestion, thanks.

**Review by kartben — DISMISSED** [review]
+1 for docs :) https://builds.zephyrproject.io/zephyr/pr/80247/docs/boards/adi/max78002evkit/doc/index.html

File: boards/adi/max78002evkit/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,298 @@
+.. zephyr:board:: max78002evkit
+
+MAX78002EVKIT
+#############
```
[the remark this answers is not in the document]
Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/index.rst]: Oh, I missed that. Updated again, thanks for your patience.

**Review by ttmut — DISMISSED** [review]

File: west.yml [hunk 11]
```diff
@@ -137,7 +137,7 @@ manifest:
       groups:
         - fs
     - name: hal_adi
-      revision: a3eecfde1c76d38312b94fd346c7ba9fe2661992
+      revision: pull/10/head
```
Comment (MaureenHelm) [inline west.yml:140]: zephyrproject-rtos/hal_adi#10 has now been merged, so this can be updated to the new SHA

**Review by MaureenHelm — COMMENTED** [review]
Need to update the manifest, but otherwise LGTM

File: west.yml (hunk 11 above)
Comment (hfakkiz) [reply west.yml]: Updated, thank you :)

**Review by MaureenHelm — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Review by ttmut — DISMISSED** [review]

File: boards/adi/max78002evkit/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+
+  - Color TFT Display
+  - Audio Stereo Codec Interface
+  - Digital Microphone
+  - A 8Mb QSPI ram
+
+Supported Features
+==================
+
+The ``max78002evkit/max78002/m4`` board target configuration supports the following interfaces:
```
Comment (nordicjm) [inline boards/adi/max78002evkit/doc/index.rst:129]: -configuration

Comment (hfakkiz) [reply boards/adi/max78002evkit/doc/index.rst]: Changed.

**Review by MaureenHelm — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**